core/ui/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/core/ui/exercise/ExerciseListItem.kt

```diff
@@ -79,16 +118,39 @@ private fun ExerciseTypeIcon(modifier: Modifier, exercise: Exercise) {
     Box(modifier = modifier) {
         Icon(
             modifier = Modifier
-                .fillMaxSize()
-                .padding(8.dp),
+                .fillMaxSize(),
             painter = getExerciseTypeIconPainter(exercise),
             contentDescription = null
         )
     }
 }
 
 /**
- * Displays the exercise title, the due data and the participation info. The participation info is automatically updated.
+ * Displays a rectangle next to the text to show the difficulty of the exercise.
+ */
+@Composable
+private fun DifficultyRectangle(modifier: Modifier, difficulty: Exercise.Difficulty) {
+    Box(modifier = modifier
+        .fillMaxHeight()
+        .width(10.dp)
+        .clip(RectangleShape)
```

Comment:
Why do you clip the the rectangle shape? Seems like a "no-operation" to me. If you want to set a certain shape, I suggest you apply it to the Modifier#background shape parameter.